src/App.tsx

```diff
@@ -26,70 +26,127 @@ type ReorderOptions = {
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends Component<{}, ReorderOptions> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  // Why when I write an arrow function everything works,
+  // but when I add a method it destroys everything. It's make me cry, dude..
+
+  doReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  doSortAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  doSortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  doReset = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const useGoods = (
+      getReorderedGoods(goodsFromServer, { sortType, isReversed })
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPHABET
+              ? 'button is-info pulse'
+              : 'button is-info is-light'}
+            onClick={this.doSortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={sortType === SortType.LENGTH
+              ? 'button is-success pulse'
+              : 'button is-success is-light'}
+            onClick={this.doSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={isReversed
+              ? 'button is-warning pulse'
+              : 'button is-warning is-light'}
+            onClick={this.doReverse}
+          >
+            Reverse
+          </button>
+
+          {sortType !== SortType.NONE || isReversed
+            ? (
+              <button
+                type="button"
+                onClick={this.doReset}
+                className="button pulse is-danger"
+              >
+                Reset
+              </button>
+            )
+            : ''}
```

Comment:
Logical operator && can be used here as you want to conditionally include an element.

Reply:
good point